test/com/facebook/buck/jvm/java/JarBuilderTest.java

```diff
@@ -85,53 +86,57 @@ public void testSortsEntriesFromAllContainers() throws IOException {
   @Test
   public void testMergesServicesFromAllContainers() throws IOException {
     File tempFile = temporaryFolder.newFile();
-    try (TestJarEntryContainer container1 = new TestJarEntryContainer("Container1");
-        TestJarEntryContainer container2 = new TestJarEntryContainer("Container2");
-        TestJarEntryContainer container3 = new TestJarEntryContainer("Container3")) {
-      new JarBuilder()
-          .addEntryContainer(
-              container1.addEntry("META-INF/services/com.example.Foo1", "com.example.Bar2"))
-          .addEntryContainer(
-              container2
-                  .addEntry("META-INF/services/com.example.Foo1", "com.example.Bar1")
-                  .addEntry("META-INF/services/com.example.Foo2", "com.example.Bar3")
-                  .addEntry("META-INF/services/com.example.Foo2", "com.example.Bar4"))
-          .addEntryContainer(
-              container3
-                  .addEntry("META-INF/services/com.example.Foo2", "com.example.Bar3")
-                  .addEntry("META-INF/services/foo/bar", "bar"))
-          .createJarFile(tempFile.toPath());
-    }
 
-    try (JarFile jarFile = new JarFile(tempFile)) {
+    for (boolean shouldDisallowAllDuplicates : Arrays.asList(true, false)) {
```

Comment:
nit: how about parameterized tests? https://github.com/junit-team/junit4/wiki/parameterized-tests :)

Reply:
Sure. I can make the change